docs/internal/GeneralArchitectureGuide.md

```diff
@@ -68,6 +68,43 @@ state must ever be reloaded from persisted state.
 
 ## Deprecations
 
+## Backwards Compatibility
+
+Elasticsearch has major versions (e.g. 8.0.0), minor versions (e.g. 8.3.0) and patch versions (e.g. 8.3.1). Multiversion
```

Comment:
I think I'd describe this a little differently. Each of 8.0.0, 8.3.0 and 8.3.1 specifies an exact release version. They all have the same major version (8) and the last two have the same minor version (8.3).